CONTRIBUTING.md

```diff
@@ -137,9 +144,61 @@ The [packages/](packages/) directory contains the following public/published pac
 ### Internal packages
 
 The [packages/](packages/) directory contains the following internal packages:
+
 * [caliper-publish](packages/caliper-publish/): Utility CLI for publishing Caliper to NPM and DockerHub.
 * [caliper-tests-integration](packages/caliper-tests-integration/): Collection of CI integration tests.
 
+### Testing Methodologies
+
+This section outlines the testing methodologies that this project follows, including both unit-level and integration-level testing.
+
+#### Unit Level Testing
+
+Unit testing focuses on testing individual packages in isolation. The tests are typically located within a test folder at the same level as the corresponding lib folder, mirroring the structure of the lib folder for easy navigation. This setup ensures that each component of the code is verified independently, confirming that it behaves as expected without interference from other parts of the system.
+
+#### Integration Level Testing
+
+Integration testing ensures that Caliper integrates correctly with various packages, effectively testing the functionality of the package itself. These tests are organized within the caliper-tests-integration folder, with each test suite dedicated to a specific package or module.
+
+#### Testing Dependencies and Their Use
+
+The following are the recommended testing modules for this project. While these tools are commonly used in the current codebase, other appropriate tools may be used as long as they facilitate effective testing.
+
+* [mockery](https://www.npmjs.com/package/mockery): Mockery is a simple module for mocking Node.js modules during testing. It allows you to replace real modules with mocks or stubs.
+  
+* [mocha](https://mochajs.org/): Mocha is a feature-rich JavaScript test framework that runs on Node.js and in the browser. It facilitates asynchronous testing, making it easy to write simple and flexible tests.
+
+* [chai](https://www.npmjs.com/package/chai): Chai is a BDD/TDD assertion library for Node.js and the browser. It can be paired with any JavaScript testing framework. We use it to create readable and expressive assertions.
+
+* [sinon](https://sinonjs.org/releases/v18/): Sinon is a standalone test spies, stubs, and mocks for JavaScript. It works with any test framework and integrates well with Mocha and Chai. We utilize Sinon for checking how functions are called during testing.
+
+* [sinon-chai](https://www.npmjs.com/package/sinon-chai): This library provides a set of custom assertions for using Sinon with Chai. It allows you to write more readable assertions for Sinon spies, stubs, and mocks.
+
+* [nyc](https://www.npmjs.com/package/nyc): NYC is a command-line utility for generating code coverage reports. It is often used with Mocha to ensure that tests cover as much code as possible.
+
+*[yeoman-generator](https://www.npmjs.com/package/yeoman-generator): Rails-inspired generator system that provides scaffolding for your apps
+
+#### Mandatory Tools in the Testing Pipeline
```

Comment:
This list really only applies to unit testing at the moment. I don't see they would be relevant for integration testing so we need to make that clear or restructure to subcategorise this under unit testing

Reply:
I was thinking of including the packages used in the caliper-tests-integration package, but I skipped the Hyperledger packages because I felt they were self-explanatory. I also believe the mandatory license and use strict requirements apply to JavaScript files under the integration tests. So, what do you think should be added to make the list more inclusive for integration tests?

Reply:
Integration tests may not be orchestrated by node necessarily so there isn't a guaranteed need for `use strict`. `use strict` is a general requirement for all node js code. Licensing is mandatory but that applies to everything again, but I was happy in the first instance to re-iterate that in the unit tests as you started by discussing the structure. We could re-iterate again or just point to a central place in the future. Now it looks like we are going 2 places in the main source code where a contributing.md file is now going to exist. We will need to merge the information going forward so once your PR is merged and @psankhe28 PR is merged, I think you 2 (@tunedev @psankhe28) need to work together to ensure a single Contributing.md file in the code base going forward. Sorry only just realised that this was going to happen as I only reviewed the docs PR yesterday.

Reply:
I think we can converge on merging them. To address your concerns with this PR, would sectioning the test methodologies for unit tests and pointing to an example integration test in the integration test description address your remaining concerns? I'm trying to gain clarity on what I am to change to resolve them.